test/server/http/admin_test.cc

```diff
@@ -243,9 +243,9 @@ TEST_P(AdminInstanceTest, Runtime) {
   std::unordered_map<std::string, Runtime::Snapshot::Entry> entries2{
       {"string_key", {"override", {}}}, {"extra_key", {"bar", {}}}};
 
-  ON_CALL(*layer1, name()).WillByDefault(testing::ReturnRef("layer1"));
+  ON_CALL(*layer1, name()).WillByDefault(testing::ReturnRefOfCopy(std::string{"layer1"}));
```

Comment:
I'm not super-fluent in gmock for various reasons, but this looks funny.  What's the lifetime of expression std::string{"layer1"} such that it makes sense to return a ref to it?

I'd feel better if you made std::string{"layer1"} a method-scoped const.

Reply:
We have to return a reference because the `name()` method returns `const std::string&`, but it's also nice to just pass the temporary string there for brevity. This is exactly what `ReturnRefOfCopy` is for; as opposed to `ReturnRef`, to which your suspicion applies, it will copy and hold on to the temporary, so you don't end up with a dangling reference. See https://github.com/google/googlemock/blob/master/googlemock/docs/CheatSheet.md#returning-a-value

Reply:
Ah, very cool.  Thanks.

Reply:
Forgot to add, full disclosure I just learned about `ReturnRefOfCopy` today as well! :)